data_steward/tools/clean_project_datasets.py

```diff
@@ -98,9 +124,26 @@ def get_response():
         '--name_substrings',
         nargs='+',
         dest='name_substrings',
-        help='Identifies substrings that help identify datasets to delete. '
-        'A dataset containing any of these substrings within in their dataset_id will be deleted. ',
+        help=('Identifies substrings that help identify datasets to delete. '
+              'A dataset containing any of these substrings within in their '
+              'dataset_id will be deleted.'),
         required=True)
-    args = parser.parse_args()
+    parser.add_argument('-s',
+                        '--console_log',
+                        dest='console_log',
+                        action='store_true',
+                        help='Send logs to console')
+
+    return parser.parse_args(raw_args)
+
+
+def main(raw_args=None):
+    args = get_arguments(raw_args)
+
+    pipeline_logging.configure(add_console_handler=args.console_log)
```

Comment:
I'd suggest making this `add_console_handler=True`. The datasets to delete do not show up otherwise.

Reply:
@nishanthpp93 there's a couple of ways to go about this.  

1. We can hard code it (which I am strongly against).  I don't want to hard code it because then we have to make code changes when we start changing where this code is run.  It's fine while we are manually running the code, but it becomes a nuissance if we were to ever import this into some other piece of code and call it from there.  We won't want to log everything to the console at that point.
2. We can default the parser argument to True instead of False, so we have to explicitly turn off the logging.  Then if we do import this into another module and call this main, we can turn the console logs off without requiring a code change.
3. We get used to using the command line to send the logs to the console as well as the log file.  The code is set up for easier programmatic reuse.  The datasets to delete  are being logged in the log file, even if they lines aren't sent to the console.

I generally try to develop so that there is less to change when something becomes part of an automated process.  That's why I generally use option 3.  I develop for the programmatic usage and expect myself to remember the extra command line flags.

Thoughts?

Reply:
@lrwb-aou Sure, I would prefer 3 as well, and follow that for other scripts. However, in this script, the question for the prompt does not show up. All I see is "Please press Y/n" - it does not inform about what I am consenting to, rather just demands consent - which can be dangerous. If just the logs for lines 79-85 which set up and pose the question are marked for printing to stdout regardless of the arg for console log, that works for me.

Reply:
@nishanthpp93 How about option 2 then?  I just changed the default value to be `True` for this.  So someone must explicitly turn off console loggig.

Reply:
@lrwb-aou Sure, I should've mentioned another option, which is to store the text for lines 79-85 in an f-string and passed as `get_response(question_text)` so that the `input` call prints all of the question text instead of just "Please press Y/n". Option 2 works, but can be counter-intuitive compared to the other scripts we have.

Reply:
I just realized - option 2 does not handle invariance - we need the lines 79-85 to be printed regardless of the arg for console log. This can be handled by passing the text to the `input` call like I mentioned above. I'm open to other ways to achieve this as well.

Reply:
@nishanthpp93 I made the suggested changes.  Let me know if you see anymore "gotchas".